### DISCUSSION TITLE
how to insert or set "null" values in mutations

### DISCUSSION THREAD

**Discussion by mojerro** [directus/directus#14931] [Archive]
I hope to set a foreign key to null like the code below, but it shows errors.
Does someone know how to deal with it?

```graphql
mutation MyMutation {
  update_consultations_item(data: {status: 0, student_id: {id: null}}, id: "7") {
    id
    name
    type
  }
}

```

**Comment by rijkvanzanten** [comment]
I don't think this is currently supported, as GraphQL requires the ID to be a value, and `student_id` can't be a union input type

**Reply by mojerro** [reply]
so is there any way or possible way to support deleting the foreign key? I'm now using endpoints, but it's increasing the complexity.

**Reply by rijkvanzanten** [reply]
Yeah, assuming the field is nullable, you should be able to just set the whole thing to `null` instead of `{ id: null }`:

```graphql
mutation MyMutation {
  update_consultations_item(data: {status: 0, student_id: null}, id: "7") {
    id
    name
    type
  }
}
```

**Reply by mojerro** [reply]
that's works well for me, thank you~
